k2/csrc/utils.h

```diff
@@ -412,7 +431,33 @@ void EvalWithRedirect(cudaStream_t stream, int32_t num_jobs,
                       TaskRedirect *redirect, int32_t min_threads_per_job,
                       int32_t tot_work, int32_t target_num_loops,
                       LambdaT &lambda) {
-  // TODO:
+  if (num_jobs <= 0 || tot_work <= 0) return;
```

Comment:
EvalWithDirectt() may not be that useful for host code, and in fact we may just avoid calling it for host code.  We may eventually have a version of Eval() that  is device-only, since this is a common pattern.  But we dont have to worry about this just yet.

Reply:
Added a `TODO` for now, will get back when we get other functions calling `TaskRedirect` tested.